### ISSUE TITLE
Player Crashing server on Join -- Same Player

### ISSUE THREAD

**Issue by gameitall** [PaperSpigot/Paper#9955]
### Stack trace

```
[01:38:30] [User Authenticator #1/INFO]: UUID of player sapir3 is aa5cd6bb-d9d1-421f-b4ae-ceadc6ed7005
[01:38:30] [Server thread/INFO]: sapir3[/70.59.238.240:55627] logged in with entity id 8918 at ([world]-48.5, 53.0, 490.5)
[01:38:30] [Server thread/INFO]: >> sapir3
[01:38:32] [Server thread/ERROR]: Failed to handle packet net.minecraft.network.protocol.game.PacketPlayInWindowClick@44a4a42e, suppressing error
net.minecraft.ReportedException: Container click
	at net.minecraft.world.inventory.AbstractContainerMenu.clicked(AbstractContainerMenu.java:397) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleContainerClick(ServerGamePacketListenerImpl.java:3245) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:58) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:23) ~[?:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:51) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1338) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1315) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1308) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1286) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1174) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:317) ~[paper-1.20.1.jar:git-Paper-196]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 36
	at java.util.Arrays$ArrayList.get(Arrays.java:4165) ~[?:?]
	at net.minecraft.core.NonNullList.get(NonNullList.java:44) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.world.entity.player.Inventory.getItem(Inventory.java:621) ~[?:?]
	at net.minecraft.world.inventory.AbstractContainerMenu.doClick(AbstractContainerMenu.java:604) ~[?:?]
	at net.minecraft.world.inventory.AbstractContainerMenu.clicked(AbstractContainerMenu.java:382) ~[?:?]
	... 17 more
[01:38:32] [Server thread/ERROR]: Failed to handle packet net.minecraft.network.protocol.game.PacketPlayInWindowClick@52560f44, suppressing error
net.minecraft.ReportedException: Container click
	at net.minecraft.world.inventory.AbstractContainerMenu.clicked(AbstractContainerMenu.java:397) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleContainerClick(ServerGamePacketListenerImpl.java:3245) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:58) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:23) ~[?:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:51) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1338) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1315) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1308) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1385) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1170) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:317) ~[paper-1.20.1.jar:git-Paper-196]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 36
	at java.util.Arrays$ArrayList.get(Arrays.java:4165) ~[?:?]
	at net.minecraft.core.NonNullList.get(NonNullList.java:44) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.world.entity.player.Inventory.getItem(Inventory.java:621) ~[?:?]
	at net.minecraft.world.inventory.AbstractContainerMenu.doClick(AbstractContainerMenu.java:604) ~[?:?]
	at net.minecraft.world.inventory.AbstractContainerMenu.clicked(AbstractContainerMenu.java:382) ~[?:?]
	... 17 more
[01:38:32] [Server thread/ERROR]: Failed to handle packet net.minecraft.network.protocol.game.PacketPlayInWindowClick@3af664e8, suppressing error
net.minecraft.ReportedException: Container click
	at net.minecraft.world.inventory.AbstractContainerMenu.clicked(AbstractContainerMenu.java:397) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleContainerClick(ServerGamePacketListenerImpl.java:3245) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:58) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:23) ~[?:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:51) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1338) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1315) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1308) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1385) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1170) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:317) ~[paper-1.20.1.jar:git-Paper-196]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 36
	at java.util.Arrays$ArrayList.get(Arrays.java:4165) ~[?:?]
	at net.minecraft.core.NonNullList.get(NonNullList.java:44) ~[paper-1.20.1.jar:git-Paper-196]
	at net.minecraft.world.entity.player.Inventory.getItem(Inventory.java:621) ~[?:?]
	at net.minecraft.world.inventory.AbstractContainerMenu.doClick(AbstractContainerMenu.java:604) ~[?:?]
	at net.minecraft.world.inventory.AbstractContainerMenu.clicked(AbstractContainerMenu.java:382) ~[?:?]
	... 17 more

```

### Plugin and Datapack List

BannerMaker, BetterRTP, BuycraftX, ChatItem, Citizens, CMILib, CoreProtect, DiscordSRV, Essentials, EssentialsSpawn, FarmLimiter, FastAsyncWorldEdit, GPFlags, GriefPrevention, GSit, HeadDatabase, HolographicDisplays, HoloMobHealth, LiteBans, NexEngine, OpenInv, *PermissionsEx, PlaceholderAPI, PremiumVanish, ProtocolLib, SkBee, Skript, spark, TAB, Vault, VentureChat, ViaVersion, Votifier, VotingPlugin, WorldGuard, WorldGuardExtraFlags

There are 2 data pack(s) enabled: [vanilla (built-in)], [file/bukkit (world)]

### Paper version

This server is running Paper version git-Paper-196 (MC: 1.20.1) (Implementing API version 1.20.1-R0.1-SNAPSHOT) (Git: 773dd72)

### Other

User joined our server for the frist time with 2 other players. No Cmds where run by any player. The player was sitting at the spawn point.

After 10 seconds the Server would start spitting out the above error causing the tps to drop then crashing the server.
After IP banning the user. The server Stabalized and is no longer having issues.

Users IP: 70.59.238.240
Username: Sapir3

Server had no changed before this time and was running smoothly for 2 weeks.

**Comment by Malfrador** [comment]
Thats an exploit that was fixed in 1.20.2. Updating will fix it.

**Comment by gameitall** [comment]
Thanks, must have missed that patch log.
Will update

On Nov 19, 2023, at 9:23 PM, Malfrador ***@***.***> wrote:

﻿

Closed #9955<https://github.com/PaperMC/Paper/issues/9955> as not planned.
